### DISCUSSION TITLE
Add general approach to send form snippets to client

### DISCUSSION THREAD

**Discussion by ovenum** [tobimori/kirby-dreamform#61] [Feature Requests]
While HTMX support in DreamForm is a really nice feature it’s not as useful if HTMX is not part of a sites frontend stack : )
Personally i use AlpineAjax for components where a similar behaviour is required.

It’s currently possible to re-create the same behaviour with AlpineAjax and using _htmx_ mode,  i would love to see DreamForm implementing a more general approach to just send the form snippet, instead of the whole page, back to the client when a form is send to the server.

This could break the current out-of-the-box behaviour for HTM, _cookbook_ articles could add documentation on how such behaviour can be realised with DreamForm.

**Comment by tobimori** [comment]
From what I've gathered, Alpine Ajax does not have an equivalent for some features required for the Htmx integration, so this isn't a simple addition.

**Reply by ovenum** [reply]
The most lacking feature in AlpineAjax was a missing equivalent to HTMX `hx-vals` attribute. I’ve created a simple test AlpineAjax DreamForm integration for a site i’m planing to use it. I did not check for multi-step support and proper testing is still required. 

The solution to the currently missing `hx-vals` attribute is an Alpine Form component that will append the values to the request issued by AlpineAjax.
